extensions/barrage/src/main/java/io/deephaven/extensions/barrage/chunk/array/ObjectArrayExpansionKernel.java

```diff
@@ -82,7 +80,7 @@ public <T, A extends Any> WritableObjectChunk<T, A> contract(
         int lenRead = 0;
         for (int i = 0; i < itemsInBatch; ++i) {
             final int rowLen = perElementLengthDest.get(i + 1) - perElementLengthDest.get(i);
-            final Object[] row = (Object[]) Array.newInstance(componentType, rowLen);
+            final Object[] row = new Object[rowLen];
```

Comment:
Right. I haven't noticed a change in behavior from this, but it is causing a failure in the FlightMessageRoundTripTest - ColumnSource.get(long) is returning an Object[] rather than the specified object type.

I'll need to find some way to delegate this, since the JS impl won't care what kind of array it gets here.

Reply:
Finally revisited. Two basic options:
 * Copy the entire class to change this one line - the JS API won't care if the array instance is the wrong type
 * Make a new util method that will only exist on the server, with a body like

    ```java
    public static Object[] makeArray(Class<?> componentType, int rowLen) {
        return (Object[]) Array.newInstance(componentType, rowLen);
    }
    ```
    This ideally would be in its own simple class, so that the client code can copy the class and replace the implementation with something specific for a browser (just `return new Object[rowLen];`).

I don't see another nice way to provide both implementations, without having some kind of ArrayFactory type that gets passed in ChunkReaderFactory -> VarListChunkReader -> ArrayExpansionKernel -> ObjectArrayExpansionKernel, which seems not worth it for just this one method.

Reply:
GIven those choices, I'd take "make a new util method".